src/components/loading-button/loading-button.types.ts

```diff
@@ -0,0 +1,29 @@
+export enum LoadingResult {
+    /**
+     * Indicate loading successful
+     */
+    SUCCESS = 'success',
+
+    /**
+     * Indicate loading unsuccessful
+     */
+    FAILURE = 'failure',
+
+    /**
+     * Stop loading indication, without indicating either success or failure.
+     * This might be used if the user cancels the loading, for example.
+     */
+    ABORT = 'abort',
+}
+
+export interface LoadingButtonInteraction {
```

Comment:
Hmm… I think this comment has just about started to win me over… 😊

I don't like adding a `loadingFailed` prop to `limel-button`, but maybe adding a `resolveLoading` _method_ would be the ticket… we could (perhaps; at least I hope we could) still use the "success" resolution if the `loading`-property is just flipped to `false`, instead of the method being called.

Thoughts? 🤔

Reply:
Good 😄
I'm starting to think a `resolveLoading` method might better than something like

```javascript
button.loading = save()
```

which is definitely my sort of magic, but maybe not the clearest interface.

I think that "abort" (not indicating success/failure) might be better if we can break with the current behaviour, but it's not super important to me.

Reply:
> ```js
> button.loading = save()
> ```
> 
> which is definitely my sort of magic, but maybe not the clearest interface.

😅

Yeah, I get the appeal… but I'd maybe go with `button.indicateLoading(save())` in that case. It would do the exact same thing, it's just more obvious that the button is gonna _do_ something with the return value of `save()`, rather than just hold on to it.